server/task_manager.py

```diff
@@ -0,0 +1,154 @@
+from pathlib import Path
+import json
+from typing import Dict, List, Optional, Any
+
+from pyserini.search.lucene import LuceneSearcher
+from pyserini.index.lucene import LuceneIndexReader
+from pyserini.prebuilt_index_info import TF_INDEX_INFO
+from pyserini.util import check_downloaded
+
+from models import IndexConfig, IndexType
+
+
+class TaskManager:
+    """Core functionality manager."""
+
+    def __init__(self):
+        self.indexes: Dict[str, IndexConfig] = {}
+
+    def initialize_default_index(self, default_index: Optional[str]) -> None:
+        """Initialize default prebuilt index."""
+
+        if default_index and default_index in TF_INDEX_INFO.keys():
+            self.add_index(
+                IndexConfig(
+                    name=default_index,
+                    type=IndexType.PREBUILT,
+                    path=default_index,
+                    description=TF_INDEX_INFO[default_index].get("description", ""),
+                )
+            )
+
+    def add_index(self, config: IndexConfig) -> None:
+        """Add a new index to the manager."""
+        if config.type == IndexType.LOCAL:
+            if not Path(config.path).exists():
+                raise FileNotFoundError(f"Index path does not exist: {config.path}")
+            config.searcher = LuceneSearcher(config.path)
+            config.reader = LuceneIndexReader(config.path)
+        else:
+            config.searcher = LuceneSearcher.from_prebuilt_index(config.path)
+            config.reader = LuceneIndexReader.from_prebuilt_index(config.reader)
```

Comment:
@lintool is this how you deal with this stuff rn

Reply:
good enough for now...